common/djangoapps/util/milestones_helpers.py

```diff
@@ -313,6 +314,28 @@ def remove_content_references(content_id):
     return milestones_api.remove_content_references(content_id)
 
 
+def any_unfulfilled_milestones(course_id, user_id):
+    """ Returns a boolean if user has any unfulfilled milestones """
+    if not settings.FEATURES.get('MILESTONES_APP', False):
+        return None
+    from milestones.exceptions import (
+        InvalidMilestoneRelationshipTypeException,
+        InvalidCourseKeyException,
+        InvalidUserException,
+    )
+    try:
+        return bool(
+            get_course_milestones_fulfillment_paths(course_id, {"id": user_id})
+        )
+    # Exception when milestone is not found
+    except InvalidMilestoneRelationshipTypeException:
+        return False
+    except InvalidCourseKeyException:
```

Comment:
What is the rationale for catching these additional exceptional exceptions?  If we are not expecting them in the usual case, we do not need to handle them specially.